### PR TITLE
fix: payment amount showing in foreign currency

### PR DESCRIPTION
**Pull request by nextchamp-saqib**
For a multicurrency invoice, the "payment amount" under "payment schedule" table has value stored in company currency, but the currency symbol is of foreign currency

image

### CODE REVIEW

File: erpnext/controllers/accounts_controller.py [hunk 1]
```diff
@@ -923,30 +923,36 @@ def set_payment_schedule(self):
...
-				data = dict(due_date=due_date, invoice_portion=100, payment_amount=grand_total)
+				data = dict(due_date=due_date, invoice_portion=100, payment_amount=grand_total, base_payment_amount=base_grand_total)
 				self.append("payment_schedule", data)
 		else:
 			for d in self.get("payment_schedule"):
 				if d.invoice_portion:
 					d.payment_amount = flt(grand_total * flt(d.invoice_portion / 100), d.precision('payment_amount'))
+					d.base_payment_amount = flt(base_grand_total * flt(d.invoice_portion / 100), d.precision('payment_amount'))
 					d.outstanding = d.payment_amount
+					d.base_outstanding = d.base_payment_amount
```
Comment (deepeshgarg007) [inline erpnext/controllers/accounts_controller.py]: Remove this

File: erpnext/controllers/accounts_controller.py [hunk 2]
```diff
@@ -1261,10 +1274,11 @@ def get_payment_term_details(term, posting_date=None, grand_total=None, bill_dat
 	term_details.description = term.description
 	term_details.invoice_portion = term.invoice_portion
 	term_details.payment_amount = flt(term.invoice_portion) * flt(grand_total) / 100
+	term_details.base_payment_amount = flt(term.invoice_portion) * flt(base_grand_total) / 100
 	term_details.discount_type = term.discount_type
 	term_details.discount = term.discount
-	# term_details.discounted_amount = flt(grand_total) * (term.discount / 100) if term.discount_type == 'Percentage' else discount
 	term_details.outstanding = term_details.payment_amount
+	term_details.base_outstanding = term_details.base_payment_amount
```
Comment (deepeshgarg007) [inline erpnext/controllers/accounts_controller.py]: This too

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

File: erpnext/controllers/accounts_controller.py [hunk 3]
```diff
@@ -982,22 +988,29 @@ def validate_payment_schedule_amount(self):
...
-			if total != flt(grand_total, self.precision("grand_total")):
+
+			if self.get("total_advance"):
+				if party_account_currency == self.company_currency:
+					base_grand_total -= self.get("total_advance")
+					grand_total = flt(base_grand_total / self.get("conversion_rate"), self.precision("grand_total"))
+				else:
+					grand_total -= self.get("total_advance")
+					base_grand_total = flt(grand_total * self.get("conversion_rate"), self.precision("base_grand_total"))
+			print(total, base_total)
```
Comment (deepeshgarg007) [inline erpnext/controllers/accounts_controller.py]: Remove print statements

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

File: erpnext/controllers/accounts_controller.py (hunk 1 above)
Comment (nextchamp-saqib) [inline erpnext/controllers/accounts_controller.py]:
```suggestion
```

File: erpnext/controllers/accounts_controller.py (hunk 3 above)
Comment (nextchamp-saqib) [inline erpnext/controllers/accounts_controller.py]:
```suggestion
```

File: erpnext/controllers/accounts_controller.py (hunk 2 above)
Comment (nextchamp-saqib) [inline erpnext/controllers/accounts_controller.py]:
```suggestion
```

**Review by deepeshgarg007 — APPROVED** [review]

**Pull request merged**